src/plans/proximal_gradient_plan.jl

```diff
@@ -126,6 +126,87 @@ function get_proximal_map!(
     mpgo.proximal_map_h!!(M, q, λ, p)
     return q
 end
+# 
+# Nonsmooth cost, for subproblem formulation
+@doc raw"""
+    ProximalGradientNonsmoothCost{F, R, P}
+
+Stores the nonsmooth part of the proximal gradient objective and the stepsize parameter ``λ ∈ ℝ``.
+
+This struct is also a functor `(M, p) -> v` that can be used as a cost function within a solver, primarily for solving the proximal map subproblem formulation in the proximal gradient method.
+
+## Fields
+* `cost::F` - the nonsmooth part of the proximal gradient objective, i.e. the part of the objective whose proximal map is sought
+* `λ::R` - the stepsize parameter for the proximal map
+* `proximity_point::P` - point where the proximal map is evaluated
```

Comment:
Hm, I am not so sure what this refers to, the argument of the prox? One could maybe at least mention that in the docs, or even the full formulae above.

Reply:
I was first thinking about changing the name, but if we thoroughly document the cost as a math formula in the disc, we can explain what this variable refers to (i.e. this fixed point from the argument that we have in the dist to the variable p).